### PR TITLE
Manual transactions

### PR DESCRIPTION
**Pull request by lexod**

### CODE REVIEW

File: app/Listeners/Deals/KickoffDeal.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            'user_id',
+            'parent_id',
+            'file_name',
+            'status',
+            'internal_status',
+            'r_code',
+            'created_at',
+            'updated_at',
+
+        ];
```
Comment (mabbus) [inline app/Listeners/Deals/KickoffDeal.php:40]: are we ever going to use r_code or file_name?

File: resources/views/employees/advances/show.blade.php [hunk 2]
```diff
@@ -288,6 +299,30 @@
...
 
+            },
+            createTransaction: function (event) {
+                event.preventDefault();
+                if(!this.data.deal.fund_status) {
+                    return;
+                }
+
+                // todo: error messages
+                if (!this._data.transactionModalData.amount || this._data.transactionModalData.amount < 1
```
Comment (mabbus) [inline resources/views/employees/advances/show.blade.php:310]: this.transctionModalData is the vue way of grabbing from "data"

File: app/Http/Controllers/Web/EmployeesDealsController.php [hunk 3]
```diff
@@ -141,17 +148,20 @@ public function createInitialTransaction(CreateInitialTransaction $request, Deal
         $data = $request->validated();
 
         throw_if($deal->transactions()->where('type',
```
Comment (mabbus) [inline app/Http/Controllers/Web/EmployeesDealsController.php:151]: this always returns error

File: app/Listeners/Deals/KickoffDeal.php (hunk 1 above)
Comment (lexod) [reply app/Listeners/Deals/KickoffDeal.php:40]: probably not

File: app/Http/Controllers/Web/EmployeesDealsController.php [hunk 4]
```diff
@@ -81,7 +85,10 @@ public function show(GetDeal $request, Deal $deal)
                 'documents' => function ($query) {
                     $query->latest();
                 },
-                'overrides'
+                'overrides',
+                'log' => function ($query) {
+                    $query->latest();
+                }
```
[the remark this answers is not in the document]
Comment (lexod) [reply app/Http/Controllers/Web/EmployeesDealsController.php:91]: latest() does sorting by created_at in DESC order

File: app/Http/Requests/Employee/Deals/CreateManualTransaction.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'transactionType' => 'required|string|in:'. implode(',', [Transaction::TYPE_MERCHANT_CREDIT, Transaction::TYPE_MERCHANT_DEBIT]),
+            'verified' => 'required|boolean',
```
[the remark this answers is not in the document]
Comment (lexod) [reply app/Http/Requests/Employee/Deals/CreateManualTransaction.php:22]: true/false/1/0/'1'/'0'
php is very awkward when it comes to boolean parsing in requests

File: app/Http/Controllers/Web/EmployeesDealsController.php (hunk 3 above)
Comment (lexod) [reply app/Http/Controllers/Web/EmployeesDealsController.php:151]: Set the wrong type

**Pull request merged**